apps/settings/js/modules/addon_manager.js

```diff
@@ -205,71 +205,114 @@ define(function(require) {
         return Promise.reject('not an addon');
       }
 
-      var manifest = this._getManifest(addon);
-      var customizations = manifest && manifest.customizations;
+      return AppsCache.apps().then(apps => Promise.all(apps.map(app =>
+        this._getCustomizedApp(addon, app)))).then(customizedApps =>
+          customizedApps.filter(customizedApp => customizedApp));
+    },
+
+    /**
+     * Returns an object, if available, describing the app if it is affected by
+     * a given addon. It has an 'app' property that holds the app object, an
+     * 'addon' property that holds the addon object, and a 'customizations'
+     * property that is an array of customizations (from the addon manifest).
+     * We implement getAddonTargets() and filterByApp() using this information
+     * and also derive the 'reboot' and 'restart' return values of
+     * disableAddon() from it.
+     *
+     * @access private
+     * @memberOf AddonManager.prototype
+     * @param {DOMApplication} addon
+     * @param {DOMApplication} app
+     * @returns {Promise}
+     */
+    _getCustomizedApp: function(addon, app) {
```

Comment:
This function can be considered as a utility function exposed to others for checking if there are any customization acting on an app. Then we are able to use it in our filter.